video/out/gpu/context.c

```diff
@@ -182,6 +186,11 @@ struct ra_ctx *ra_ctx_create(struct vo *vo, const char *context_type,
         };
 
         MP_VERBOSE(ctx, "Initializing GPU context '%s'\n", ctx->fns->name);
+        if (strstr(ctx->fns->name, "wayland") && gnome && vo->probing) {
+            MP_WARN(ctx, "GNOME's wayland compositor is known to have many serious issues with mpv. Switch to GNOME's xorg session for the best experience.\n");
+            talloc_free(ctx);
+            continue;
+        }
```

Comment:
Next problem I realise. This warning still shows when using GNOME on X11, because it happens at the beginning of probing, so it doesn't care if Wayland is actually there and used or not.

I think, given that GNOME is crap whether you use Wayland or XWayland, that we shouldn't actually abort the probe, and instead just let it load, in which case the warning in wayland_common is all we need.

Reply:
The only thing is that there won't be any warning if someone uses `--gpu-context=x11egl/x11`. But I guess it's probably OK.